hack/prom_metric_linter.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+
+#
+# This file is part of the KubeVirt project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2023 Red Hat, Inc.
+#
+#
+set -e
+
+linter_image_tag="v0.0.1"
+
+PROJECT_ROOT="$(readlink -e "$(dirname "${BASH_SOURCE[0]}")"/../)"
+export METRICS_COLLECTOR_PATH="${METRICS_COLLECTOR_PATH:-${PROJECT_ROOT}/tools/prom-metrics-collector}"
+
+if [[ ! -d "$METRICS_COLLECTOR_PATH" ]]; then
+    echo "Invalid METRICS_COLLECTOR_PATH: $METRICS_COLLECTOR_PATH is not a valid directory path"
+    exit 1
+fi
+
+# Parse command-line arguments
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+    --operator-name=*)
+        operator_name="${1#*=}"
+        shift
+        ;;
+    --sub-operator-name=*)
+        sub_operator_name="${1#*=}"
+        shift
+        ;;
+    *)
+        echo "Invalid argument: $1"
+        exit 1
+        ;;
+    esac
+done
+
+# Get the metrics list
+go build -o _out/prom-metrics-collector "$METRICS_COLLECTOR_PATH/..."
+json_output=$(_out/prom-metrics-collector 2>/dev/null)
+
+# Run the linter by using the prom-metrics-linter Docker container
+errors=$(docker run -i "quay.io/kubevirt/prom-metrics-linter:$linter_image_tag" \
```

Comment:
@orenc1 @nunnatsa do we have a parameter for docker/podman I used docker for no special reason.

Reply:
> @orenc1 @nunnatsa do we have a parameter for docker/podman I used docker for no special reason.

Yes: e.g. here: https://github.com/kubevirt/hyperconverged-cluster-operator/blob/1064220644c06831cb625ce09e8f598c68671260/hack/upgrade-test-clusterserviceversion.sh#L15-L17

Reply:
@nunnatsa I was able to implement but output added since this change:

before output is:
```
[alitman@alitman hyperconverged-cluster-operator]$ make lint-metrics
./hack/prom_metric_linter.sh --operator-name="kubevirt" --sub-operator-name="hco"
```

Now output is:
```
[alitman@alitman hyperconverged-cluster-operator]$ make lint-metrics
./hack/prom_metric_linter.sh --operator-name="kubevirt" --sub-operator-name="hco"
++ podman ps
hack/cri-bin.sh: line 5: podman: command not found
++ docker ps
++ _cri_bin=docker
++ echo 'selecting docker as container runtime'
selecting docker as container runtime
++ CRI_BIN=docker
++ CRI_INSECURE=
++ docker run -i quay.io/kubevirt/prom-metrics-linter:v0.0.1 '--metric-families=[{"name":"kubevirt_hco_out_of_band_modifications_total","help":"Count of out-of-band modifications overwritten by HCO","type":0},{"name":"kubevirt_hco_number_of_unsafe_modifications","help":"Count of unsafe modifications in the HyperConverged annotations","type":1},{"name":"kubevirt_hco_hyperconverged_cr_exists","help":"Indicates whether the HyperConverged custom resource exists (1) or not (0)","type":1},{"name":"kubevirt_hco_system_health_status","help":"Indicates whether the system health status is healthy (0), warning (1), or error (2), by aggregating the conditions of HCO and its secondary resources","type":1}]' --operator-name=kubevirt --sub-operator-name=hco
+ errors=
+ [[ '' != '' ]]
```

Do you know why this output added?